test/controllers/projects_controller_test.rb

```diff
@@ -197,31 +308,85 @@ def test_edit_project
     assert_form_action(action: :update, id: project.id.to_s)
   end
 
+  def test_edit_project_any_date
+    project = projects(:bolete_project)
+    assert_true(project.start_date.nil? && project.end_date.nil?,
+                "Test needs Project with nil start and end dates")
+    params = { id: project.id.to_s }
+
+    login(project.user.login)
+    post(:edit, params: params)
+
+    assert_select("#project_dates_any_true[checked]", true,
+                  "'Any' radio button should be selected")
+  end
+
+  def test_edit_project_empty_name
+    edit_project_helper("", projects(:eol_project))
+  end
+
+  def test_edit_project_existing
+    edit_project_helper(projects(:bolete_project).title,
+                        projects(:eol_project))
+  end
+
   def test_update_project
     title = "EOL Project"
     summary = "This has become the Entoloma On Line project"
     project = Project.find_by(title: title)
     assert(project)
     assert_not_equal(summary, project.summary)
     assert_not(project.open_membership)
-    params = build_params(title, summary)
+    start_date = Time.zone.today
+    end_date = start_date + 4.days
+    params =
+      build_params(title, summary,
+                   start_date: start_date, end_date: end_date)
+
     params[:project][:open_membership] = true
     params[:id] = project.id
+
     put_requires_user(:update, { action: :show }, params)
+
     project = Project.find_by(title: title)
     assert_redirected_to(project_path(project.id))
     assert(project)
     assert_equal(summary, project.summary)
     assert(project.open_membership)
+    assert_equal(start_date, project.start_date)
+    assert_equal(end_date, project.end_date)
   end
 
-  def test_edit_project_empty_name
-    edit_project_helper("", projects(:eol_project))
-  end
+  def test_update_project_end_before_start
+    proj = projects(:pinned_date_range_project)
+    start_date = Time.zone.today
+    end_date = Time.zone.yesterday
+    params = {
```

Comment:
Again seems very similar to `build_params`.  Might need to make `build_params` a bit smarter, but I think it would be worth it.